cmd/promoted-image-governor/main.go

```diff
@@ -0,0 +1,180 @@
+	kubeconfigFlagDescription := "Path to the kubeconfig file to use for CLI requests."
+	if f := fs.Lookup("kubeconfig"); f != nil {
+		f.Usage = kubeconfigFlagDescription
+		// https://i.kym-cdn.com/entries/icons/original/000/018/012/this_is_fine.jpeg
+		defer func() { opts.kubeconfig = f.Value.String() }()
+	} else {
+		fs.StringVar(&opts.kubeconfig, "kubeconfig", "", kubeconfigFlagDescription)
+	}
+	fs.StringVar(&opts.ciOperatorconfigPath, "ci-operator-config-path", "", "Path to the ci operator config")
+	fs.BoolVar(&opts.dryRun, "dry-run", true, "Whether to run the controller-manager with dry-run")
+	fs.Var(&opts.ignoredImageStreamTagsRaw, "ignored-image-stream-tags", "An imagestreamtag to ignore while checking. It must be in namespace/name:tag format (e.G `ci/clonerefs:latest`). Can be passed multiple times.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("could not parse input")
+	}
+	return opts
+}
+
+func (o *options) validate() error {
+	if o.ciOperatorconfigPath == "" {
+		return fmt.Errorf("--ci-operator-config-path must be set")
+	}
+	return nil
+}
+
+func tagsToPromote(cfg *api.ReleaseBuildConfiguration) sets.String {
+	tags := sets.NewString()
+	for _, tag := range release.PromotedTags(cfg) {
+		tags.Insert(fmt.Sprintf("%s/%s:%s", tag.Namespace, tag.Name, tag.Tag))
+	}
+	return tags
+}
+
+func addSchemes() error {
+	if err := imagev1.AddToScheme(scheme.Scheme); err != nil {
+		return fmt.Errorf("failed to add imagev1 to scheme: %w", err)
+	}
+	if err := buildv1.AddToScheme(scheme.Scheme); err != nil {
+		return fmt.Errorf("failed to add buildv1 to scheme: %w", err)
+	}
+	return nil
+}
+
+func main() {
+	logrusutil.ComponentInit()
+
+	opts := newOpts()
+
+	if err := opts.validate(); err != nil {
+		logrus.WithError(err).Fatal("failed to validate the option")
+	}
+
+	if err := addSchemes(); err != nil {
+		logrus.WithError(err).Fatal("failed to add schemes")
+	}
+
+	abs, err := filepath.Abs(opts.ciOperatorconfigPath)
+	if err != nil {
+		logrus.WithError(err).WithField("opts.ciOperatorconfigPath", opts.ciOperatorconfigPath).Fatal("failed to determine absolute filepath")
+	}
+	promotedTags := sets.NewString()
+	if err := config.OperateOnCIOperatorConfigDir(abs, func(cfg *api.ReleaseBuildConfiguration, metadata *config.Info) error {
+		promotedTags.Insert(tagsToPromote(cfg).List()...)
+		return nil
+	}); err != nil {
+		logrus.WithField("path", abs).Fatal("failed to operate on CI Operator's config directory")
+	}
+
+	kubeConfigs, _, err := util.LoadKubeConfigs(opts.kubeconfig, nil)
+	if err != nil {
+		logrus.Fatalf("could not load kube config from path %s: %w", opts.kubeconfig, err)
+	}
+	if len(kubeConfigs) != 1 {
+		logrus.Fatalf("found %d contexts in kube config %s: must be %s only", len(kubeConfigs), opts.kubeconfig, string(api.ClusterAPPCI))
+	}
+
+	client, err := ctrlruntimeclient.New(kubeConfigs[string(api.ClusterAPPCI)], ctrlruntimeclient.Options{})
```

Comment:
We should not pass this tool a kubeconfig with multiple contexts, and we should not load multiple contexts only to pick out one specific one. There's no need for the context to have a specific name.

Reply:
Updated.